.github/workflows/pyinstaller_gui_mac.yaml

```diff
@@ -1,25 +1,40 @@
 name: Compile GUI for Mac
-# run this job manually
-on: workflow_dispatch
+on:
+  push:
+    tags:
+      # Push events to every tag including hierarchical tags like v1.0/beta
+      - '**'
+
+# Allows you to run this workflow manually too
+  workflow_dispatch:
+
 jobs:
   deploy:
     runs-on: macos-latest
     steps:
-      # used to checkout different branches
+      # used to check out different branches
       - uses: actions/checkout@v3
-        with:
-          # checkout my specific branch
-          ref: main
+#        with:
+#          # checkout my specific branch
+#          ref: master
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
       - name: pip install dependencies
-        run: pip install -r requirements.txt
+        run: pip install -r requirements_dev.txt
```

Comment:
Naïve question: What's the purpose of `requirements` if you always need `requirements_dev` when you build either locally or on the CI? Do Users have to install it using python?

Reply:
Originally when the Excel uploader was made, users would pull from master and install requirements.txt, and requirements_dev.txt was then separate for us developers when we were working on it.

With the GUI executable users no longer need requirements.txt because everything comes in the executable. I wanted to combine requirements.txt and requirements_dev.txt into one, but I was told to keep them separate, which is the reason why we have 2 requirements files